### DISCUSSION TITLE
How to replace the default istio CA with a plug-in certificate without redeploying

### DISCUSSION THREAD

**Discussion by fizos** [istio/istio#50416] [Q&A] [answered]
I have deployed  the  istio.  And now I want to replace its default certificate with a third-party plug-in certificate, do I have to redeploy istiod? Is it possible without redeploying? Help need, thanks !

**Comment by zirain** [comment]
hope [this](https://github.com/zirain/istio-root-cert-rotation/tree/main/kubecon-eu-2024) will help you.

**Reply by fizos** [reply]
Sorry the link return 404 not found

**Reply by zirain** [reply]
sorry, I forgot to make it public, it should be OK now.

**Reply by fizos** [reply]
This link needs to obtain the initial certificate A when the certificate is rotated. My cluster currently uses the default certificate and includes already running workloads. How should I replace it with a third-party certificate in this scenario? My istio version It is 17.3, deployed using helm

**Reply by zirain** [reply]
what does default certificate means? the selfsigned certificate created by istio?

**Reply by fizos** [reply]
yes,  the selfsigned certificate created by istio

**Reply by zirain** [reply, marked as answer]
maybe you can try to copy it into cacert first, then follow the rest steps.

**Reply by fizos** [reply]
According to the link steps, I was able to roll the CA from A to B normally, but when using the self-signed CA generated by istio by default as the initial state, I switched it to CA A. From istiod, the multi-root CA configuration did not take effect. , the certificate of the workload is not updated, what is the reason?
![image](https://github.com/istio/istio/assets/96819554/908affd3-502a-4bab-bf18-9d08a9371273)

**Reply by fizos** [reply]
According from the current information, when istio is deployed, if cacerts exist, istio will not generate istio-ca-secret objects. Whether it is multi-root configuration only focuses on changes to cacerts objects. If istio-ca-secret exists, it will be ignored?
